### PR TITLE
adding rs heater large

### PR DESCRIPTION
**Pull request by bogguard**
This adds the rs heater of the large home appliances.
It currently supports lock, temp_set, current_temp and switch.

### CODE REVIEW

**Comment by bogguard** [comment]
Tested it succesfully with the W'EAU Pool Heater pump

File: drivers/heater-large/device.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    if (typeof status['switch'] === 'boolean') {
+      this.setCapabilityValue('onoff', status['switch']).catch(this.error);
+    }
+
+    if (typeof status['temp_current'] === 'number') {
+      this.setCapabilityValue(
+        'measure_temperature',
+        status['temp_current'] > 100 || status['temp_current'] < 100
+          ? status['temp_current'] / 10
+          : status['temp_current']
```
Comment (bobvandevijver) [inline drivers/heater-large/device.js:49]: Can you explain this one?

**Review by bobvandevijver — COMMENTED** [review]
Thank you for the PR! Unfortunately, I cannot merge this as is, which has two reasons:

1. We are working on a Typescript migration, see #134. This would need to be adjusted as well, but the best would be to wait for #134 to be merged (which will be done this week)
2. We are aiming to get as less drivers as possible, so having a large variant of a heater does not match with that. I do believe this can be merged with the current heater, by mapping the additional Tuya capabilities with our own.

Also, can you please share the device specification? That helps us understand what you did and why.

**Comment by bogguard** [comment]
No problem, I'll move this one to typescript once that PR is merged.
I'll investigate if I can add the large heater to the current heater for small appliances. Unfortunately Tuya chose to name the same functions differently (lock vs child_lock, eco_mode vs eco, ...)

File: drivers/heater-large/device.js (hunk 1 above)
Comment (bogguard) [reply drivers/heater-large/device.js:49]: Yes, when testing the w'eau the value of the temp_current is between 0 and 1000 so I had to divide it by 10 to get the correct temperature. (example 247 is 24.7 degrees) 
I do believe they did this to support decimals, but couldn't find any documentation on it at Tuya.

As the Smart Life app also represents the value divided by 10 I believe they are doing kind of the same but unsure how it exactly is implemented in the Smart Life App. You had the same experience with other Tuya device types?

**Comment by bogguard** [comment]
The device specification : 

`{"active_time":1688287178,"biz_type":18,"category":"rs","create_time":1688285375,"icon":"smart/icon/ay1513418289516FAXSV/df2cf842ab71c359308c3814492d5ee9.png","id":"bfff3ce552aa537acbjj39","ip":"<redacted>","lat":"<redacted>","local_key":"<redacted>","lon":"<redacted>","model":"RCDF211344","name":"Weau pool heat pump","online":true,"owner_id":"<redacted>","product_id":"zqwet1zcvj2wcxzw","product_name":"Weau pool heat pump","status":[{"code":"switch","value":true},{"code":"mode","value":"eheat"},{"code":"temp_set","value":29},{"code":"temp_current","value":278},{"code":"fault","value":0}],"sub":false,"time_zone":"+02:00","uid":"<redacted>","update_time":1723640777,"uuid":"<redacted>"}`

**Comment by bobvandevijver** [comment]
I think it can be added by just adding new mappings for those properties, but if you like we can also look into?

File: drivers/heater-large/device.js (hunk 1 above)
Comment (bobvandevijver) [reply drivers/heater-large/device.js:49]: We have a same issue with sockets, where the power factor is also different per device. We solved that by adding a scaling setting, I think that would be a great fit here as well.

**Comment by bogguard** [comment]
closing this one and opening a new one with typescript integration

**Pull request closed without merging**